### PR TITLE
8293886: The abstract keyword can be removed in AESCipher

### PR DESCRIPTION
**Pull request by XueleiFan**
Hi,

Please review this simple fix for readability.

In the AES cipher implementation, the AESCipher class is defined as abstract.  As is not necessary as there is no abstract method in this class. Code reader may try to search for abstract methods if the abstract keyword is present.

BTW, I also added Override tags and make a few other cleanup, for example adding the 'final' keywords.

Thanks,
Xuelei

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293886](https://bugs.openjdk.org/browse/JDK-8293886): The abstract keyword can be removed in AESCipher

### Reviewers
 * [Anthony Scarpino](https://openjdk.org/census#ascarpino) (@ascarpino - **Reviewer**) ⚠️ Review applies to [defa6e5d](https://git.openjdk.org/jdk/pull/10297/files/defa6e5d8d47effc44a46899224b106e30e0cd79)
 * [Bradford Wetmore](https://openjdk.org/census#wetmore) (@bradfordwetmore - **Reviewer**) ⚠️ Review applies to [defa6e5d](https://git.openjdk.org/jdk/pull/10297/files/defa6e5d8d47effc44a46899224b106e30e0cd79)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10297/head:pull/10297` \
`$ git checkout pull/10297`

Update a local copy of the PR: \
`$ git checkout pull/10297` \
`$ git pull https://git.openjdk.org/jdk pull/10297/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10297`

View PR using the GUI difftool: \
`$ git pr show -t 10297`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10297.diff">https://git.openjdk.org/jdk/pull/10297.diff</a>

</details>

### CODE REVIEW

**Review by ascarpino — APPROVED** [review]
The change looks fine

File: src/java.base/share/classes/com/sun/crypto/provider/AESCipher.java [hunk 1]
```diff
@@ -58,13 +58,14 @@
  * @see OutputFeedback
  */
 
-abstract class AESCipher extends CipherSpi {
+class AESCipher extends CipherSpi {
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/com/sun/crypto/provider/AESCipher.java:61]: If you're doing general cleaning, line 33 import ByteBuffer can go, and there's an extra line in the javadoc at line 53 that could also go.

**Review by bradfordwetmore — APPROVED** [review]
Two minor nits, but otherwise looks good.

File: src/java.base/share/classes/com/sun/crypto/provider/AESCipher.java (hunk 1 above)
Comment (XueleiFan) [reply src/java.base/share/classes/com/sun/crypto/provider/AESCipher.java:61]: Yes. I will make the cleanup as well.

**Pull request closed without merging**